packages/@webex/plugin-cc/src/services/core/WebSocket/WebSocketManager.ts

```diff
@@ -0,0 +1,186 @@
+import {Signal} from '../Signal';
+import {WebexSDK, SubscribeRequest, HTTP_METHODS, WelcomeResponse} from '../../../types';
+import {SUBSCRIBE_API, WCC_API_GATEWAY} from '../../constants';
+import {SubscribeResponse} from '../../config/types';
+import LoggerProxy from '../../../logger-proxy';
+import workerScript from './keepalive.worker';
+
+export class WebSocketManager {
+  readonly onMessage: Signal.WithData<string>;
+  private readonly onMessageSend: Signal.Send<string>;
+  readonly onSocketClose: Signal.Empty;
+  private readonly onSocketCloseSend: Signal.SendEmpty;
+  private websocket: WebSocket;
+  shouldReconnect: boolean;
+  isSocketClosed: boolean;
+  private isWelcomeReceived: boolean;
+  private url: string | null = null;
+  private forceCloseWebSocketOnTimeout: boolean;
+  private isConnectionLost: boolean;
+  private webex: WebexSDK;
+  private welcomePromiseResolve:
```

Comment:
This was a needed change so that when we call the initWebSocket from the cc.ts, we are able to return a promise to resolve the initial registration. This differs from agentx which was calling it from the aqm-reqs and hence it could be handled via signals, however since we call it from cc.ts, I had to add this here for that.
It works fine (as shown in the GIF).